### PR TITLE
#61 - EC2 Rename-Tag Action

### PR DESCRIPTION
**Pull request by joshuaroot**
* Open new PR as issues with fork caused a loss
* Schema update
* Updated UnitTest to properly test renaming

### CODE REVIEW

File: c7n/resources/ec2.py [hunk 1]
```diff
@@ -606,6 +606,86 @@ def process_volume_set(self, resource):
...
+        """
+        Move source tag value to destination tag value
+
+        - Collect value from old tag
+        - Delete old tag
+        - Create new tag & assign stored value
+        """
+        self.log.info("Renaming tag on %s instances" % (len(resources)))
+        tag_map = self.data.get('map', {})
+        for t in tag_map:
```
Comment (kapilt) [inline c7n/resources/ec2.py]: why is map being referenced multiple times in this class, when its not something a user can set per the schema... ie its always empty.

File: c7n/resources/ec2.py [hunk 2]
```diff
@@ -652,6 +652,83 @@ def process_volume_set(self, resource):
                     response['SnapshotId']],
                 Tags=tags)
 
+@actions.register('rename-tag')
+class RenameTag(BaseAction):
+
+    schema = type_schema(
+        'rename-tag',
+        map={
```
Comment (kapilt) [inline c7n/resources/ec2.py]: sorry missed this key earlier, but still puzzled, why nest the values in a map if your just processing a single key/value versus making them top level

Comment (joshuaroot) [inline c7n/resources/ec2.py]: Perhaps I misinterpreted something you wrote in an earlier comment regarding this... I can revert this back.

**Comment by kapilt** [comment]
looks fine, one thing that seems odd is that coverage is reporting a decrease which seems odd.

**Comment by joshuaroot** [comment]
Coverage is down... let me look into that.

**Comment by joshuaroot** [comment]
Yep, need to adjust. Looks like my grouping function includes all the tags, not just the ones set in params. I need to correct this.

**Comment by kapilt** [comment]
not sure if this should be a followup or in this pull request, but given the tag api is generic across ec2 resources, making this ec2 instance specific seems silly.

- move to tags.py
- use model.id instead of InstanceId
- update query.py to add to all ec2 api resources

**Pull request closed without merging**